mithril-aggregator/src/services/cardano_transactions_importer.rs

```diff
@@ -60,47 +59,38 @@ impl CardanoTransactionsImporter {
         block_scanner: Arc<dyn BlockScanner>,
         transaction_store: Arc<dyn TransactionStore>,
         dirpath: &Path,
-        rescan_offset: Option<usize>,
         logger: Logger,
     ) -> Self {
         Self {
             block_scanner,
             transaction_store,
             logger,
-            rescan_offset,
             dirpath: dirpath.to_owned(),
         }
     }
 
-    async fn import_transactions(&self, up_to_beacon: ImmutableFileNumber) -> StdResult<()> {
-        let from = self.get_starting_beacon().await?;
+    async fn import_transactions(&self, up_to_beacon: BlockNumber) -> StdResult<()> {
+        let from = self.transaction_store.get_highest_beacon().await?;
         self.parse_and_store_transactions_not_imported_yet(from, up_to_beacon)
             .await
     }
 
-    async fn get_starting_beacon(&self) -> StdResult<Option<u64>> {
-        let highest = self.transaction_store.get_highest_beacon().await?;
-        let rescan_offset = self.rescan_offset.unwrap_or(0);
-        let highest = highest.map(|h| (h + 1).saturating_sub(rescan_offset as u64));
-        Ok(highest)
-    }
-
     async fn parse_and_store_transactions_not_imported_yet(
         &self,
-        from: Option<ImmutableFileNumber>,
-        until: ImmutableFileNumber,
+        from: Option<ChainPoint>,
```

Comment:
Do we really need the full `ChainPoint` here?
```suggestion
        from: Option<BlockNumber>,
```

Reply:
We need the full chain point for the `BlockStreamer` called by this method.

Reply:
We may use a `BlockNumber` but doing so we need to transfer the responsibility to obtain a chain point from a block number to the streamer.

This would be cumbersome since:
- The lower bound is acquired by the importer (that part of its current role), to do so it do a database round-trip.
- We would need to define a trait that take a block number as parameter and return a chain point ... implemented by doing another database round-trip (so two needed instead of one as now).

Another possibility would be to remove altogether the lower bound computation from the importer and transfer it to the scanner.

Reply:
Indeed this shoudl probably be the responsibility of the scanner. T
his can be done in a separate PR :+1: